tests/test_downloadermiddleware_retry.py

```diff
@@ -104,5 +104,86 @@ def _test_retry_exception(self, req, exception):
         self.assertEqual(req, None)
 
 
+class MaxRetryTimesTest(unittest.TestCase):
+    def setUp(self):
+        self.crawler = get_crawler(Spider)
+        self.spider = self.crawler._create_spider('foo')
+        self.mw = RetryMiddleware.from_crawler(self.crawler)
+        self.mw.max_retry_times = 2
+
+    def test_without_metakey(self):
+        
+        req = Request('http://www.scrapytest.org/invalid_url')
+
+        # SETTINGS: RETRY_TIMES is NON-ZERO
+        self.mw.max_retry_times = 5
+        self._test_retry(req, DNSLookupError('foo'), 5)
+
+        # SETTINGS: RETRY_TIMES = 0
+        self.mw.max_retry_times = 0
+        self._test_retry(req, DNSLookupError('foo'), 0)
+
+        # RESET RETRY_TIMES SETTINGS
+        self.mw.max_retry_times = 2
+
+    def test_with_metakey_preceding(self):
+        # request with meta(max_retry_times) is called first
```

Comment:
I don't understand the comment _"request with meta(max_retry_times) is called first"_ what do  you mean by that?
Also, you can use docstrings to describe what the test is about.